build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/flubTasks.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { existsSync } from "fs";
+import { LeafWithDoneFileTask } from "./leafTask";
+import path from "path";
+import { readFileAsync } from "../../../common/utils";
+import { GitRepo } from "../../../common/gitRepo";
+
+export class FlubListTask extends LeafWithDoneFileTask {
+	private getResourceGroup() {
+		const split = this.command.split(" ");
+		const index = split.indexOf("-g");
+		return index >= 0 ? split[index + 1] : undefined;
+	}
+	public async getDoneFileContent(): Promise<string | undefined> {
+		const resourceGroup = this.getResourceGroup();
+		if (resourceGroup === undefined) {
+			return undefined;
+		}
+		const packages = Array.from(this.node.buildContext.repoPackageMap.values()).filter(
+			(pkg) => pkg.monoRepo?.kind === resourceGroup,
+		);
+		if (packages.length === 0) {
+			return undefined;
+		}
+		return JSON.stringify(packages.map((pkg) => [pkg.name, pkg.packageJson]));
+	}
+}
+
+export class FlubCheckLayerTask extends LeafWithDoneFileTask {
+	private async getLayerInfoFile() {
+		const split = this.command.split(" ");
+		const index = split.indexOf("--info");
```

Comment:
note to self: there's a parse method in oclif that might do a lot of this for us and let us more easily "teach" fluid-build about flub commands. For the future.